jdbc-yugabytedb/src/main/java/org/postgresql/Driver.java

```diff
@@ -450,12 +451,89 @@ public Connection getResult(long timeout) throws SQLException {
    * thread without enforcing a timeout, regardless of any timeout specified in the properties.
    *
    * @param url the original URL
-   * @param props the parsed/defaulted connection properties
+   * @param properties the parsed/defaulted connection properties
    * @return a new connection
    * @throws SQLException if the connection could not be made
    */
-  private static Connection makeConnection(String url, Properties props) throws SQLException {
-    return new PgConnection(hostSpecs(props), user(props), database(props), props, url);
+  private static Connection makeConnection(String url, Properties properties) throws SQLException {
+    LoadBalanceProperties lbprops = new LoadBalanceProperties(url, properties);
+    if (lbprops.hasLoadBalance()) {
+      Connection conn = getConnectionBalanced(lbprops);
+      if (conn != null) {
+        return conn;
+      } else {
```

Comment:
the 'else' keyword is not needed (since the if block would return)

Reply:
It may return null if it is talking to servers which are not upgraded with the latest added yb_servers() function but still the client is trying to use the load-balanced property. It is simply warning it that 'load-balanced' attempt  failed and then it fails back to the original behaviour. It's a guard against client server incompatibility for load-balance feature.

Reply:
I didn't say else block is not needed.
You can keep the block, omitting the 'else'

This is minor, though.

Reply:
done